src/transport_ofi.c

```diff
@@ -578,48 +598,65 @@ static inline int atomic_limitations_check(void)
 
     int j;
     /* Binary OPS check */
-    for(i=0; i<3; i++) {//DT
+    for(i=0; i<4; i++) {//DT
       for(j=0; j<3; j++) { //OPS
         ret = fi_atomicvalid(shmem_transport_ofi_epfd, SHM_DT_INT[i], SHM_BOPS[j],
                         &atomic_size);
-        if(ret!=0 || atomic_size == 0) {
-            fprintf(stderr, "%s OFI detected no support for atomic '%s'"
-                    "on type '%s'\n", (atomicwarn ? "Warning" : "Error"),
-                    SHM_NAMED_BOPS[j], SHM_NAMED_DT_INT[i]);
-            if(!atomicwarn) {
-                OFI_ERRMSG("Error: atomicvalid ret=%d atomic_size=%d \n",
-                            ret, (int)atomic_size);
-	            return ret;
-            }
-        }
+         if(atomicvalid_rtncheck(ret, atomic_size, atomicwarn,
+                            SHM_NAMED_BOPS[j], SHM_NAMED_DT_INT[i]))
+           return ret;
       }
     }
 
     /* OTHER OPS check */
-    for(i=0; i<5; i++) {//DT
+    for(i=0; i<6; i++) {//DT
```

Comment:
sure, I sort of prefer a sizeof calculation in the limitations check but either is fine